### DISCUSSION TITLE
G3 EOL?

### DISCUSSION THREAD

**Discussion by IzzySoft** [hawkeye116477/waterfox-deb-rpm-arch-AppImage#46] [General]
As G3 wasn't updated anymore since 11/2021, do you recommend switching to G4 – and if so, what is the suggested upgrade path (and maybe that could be added to the Readme)? I vaguely remember it was planned to have the transition happening "automatically"; not sure if that's still the plan.

I'm using G3 via the OBS-PPA on Linux Mint 20.2, in case that matters.

**Comment by hawkeye116477** [comment]
From what I know some things in G4 weren't working as good or correctly like in G3. I don't know how it's currently, you can use even both versions, make few tests and then decide what to do.

> transition happening "automatically";

For now there is transition from G2 to G3. In the future (April?) maybe I'll make transition to G4.

PS: Honestly now I moved to Vivaldi browser, so maybe some things will change soon.

**Reply by IzzySoft** [reply]
Thanks for your fast answer! I was just wondering that I still got regular updates for Classic, but none for G3 for a while – which is why I checked. I saw releases on the WF repo stopped for G3 and "transformed" to G4 back in November – as it happens for most software projects when a new regular release is available. This led me to the conclusion there would be a "smooth transition". But downloading the latest G4 `*.deb` and looking inside rather suggests a bit different: at least the binary has a different name, so wherever it was used in one's own scripts one needs either a symlink or a "wrapper script". Which would not be a big deal, though – and immediately reminds me of `update-alternatives` and `www-browser`, so I could make a `www-waterfox` for that :stuck_out_tongue_winking_eye:

> you can use even both versions, make few tests and then decide what to do.

This raises the next question: do both share the same profiles, or has G4 its own (like it is with Classic vs. G3 – I've never used G2)? If they share, I guess there would be issues if both are running at the same time. If they don't share, would I need to configure everything again (or can the profile be "imported" – as last resort e.g. by copying the G3 one over the default profile G4 created)?

> now I moved to Vivaldi browser, so maybe some things will change soon

Oh, that's sad news for us who used the PPA you've maintained so well all the time! I must admit I felt forced to use multiple browsers simultaneously now. 3 of them are permanently open: Classic is still my main driver. G3 is open for the stuff no longer working on Classic (GitLab, Github, Stack Exchange), and Tor-Browser for… well, those sites not working without massive tracking one sometimes has to open. Number 4 is ungoogled Chromium only opened for video conferences. The web is a mess…

**Reply by hawkeye116477** [reply]
> do both share the same profiles, or has G4 its own

They share same profile.

> Oh, that's sad news for us who used the PPA you've maintained so well all the time! 

Well, Waterfox went rather to bad direction, Classic is neglected, less security and G4 hasn't full legacy extensions support, even docs for making that extensions for G4 are incomplete and outdated. Honestly mainly I wanted great sidebar and little tweaking browser's appearance, rounded tabs and all that's possible in Vivaldi, so I moved. However I hope that in the future I could still continue using uBO.


Anyway, maybe I'll at least keep one uniwersal package (AppImage?) or delay between updates will be longer, I'll see.

**Reply by IzzySoft** [reply]
Thanks for the clear words and details!

> They share same profile.

So am I correct this means:

1. I should not have instances of G3 & G4 open at the same time, to avoid corruption
2. having used G4 to take a look, I can simply close it and switch back to G3 without any issues (i.e. there are no changes to the profile making it incompatible with G3)?

Then I could give it a try and, if it works out, simply switch over. With no more updates to G3, it would be a bad idea sticking to it for a longer time, security-wise.

> Waterfox went rather to bad direction

So sad to read those details! I hadn't time yet to set up G3 in a way it supports legacy extensions at all, or I might have switched from Classic to G3 already (I hope the other modifications will still work). Reading that with G4 this won't even be possible is very bad news: I still use several legacy exts that never got ported to webext (mostly due to lacking APIs, some also because developers simply gave up to play along with Mozilla's jumps).

> one uniwersal package (AppImage?)

I'd be open to switch to that if it became necessary (luckily you choose AppImage – I'd e.g. not be willing to switch to flatpak and definitely not to snap :see_no_evil:), though I'd prefer the PPA. And while delayed updates certainly are no "feature", I could live with them, too, if they're not *too* big (a few weeks are fine, a few months.. well, not so :rofl:)

Thanks again for all the time you put into this, @hawkeye116477 – highly appreciated!
